contracts/container/contract.go

```diff
@@ -491,6 +494,67 @@ func List(owner []byte) [][]byte {
 	return list
 }
 
+// AddNextEpochNodes adds container members for the next epoch. Changes must be
+// committed with [CommitContainerListUpdate]. Without submitting, it is a no-op
+// operation. Results of the call operation can be received via [Nodes]. This
+// method must be called only when a container list is changed, otherwise
+// nothing should be done.
+// Call must be signed by the Alphabet nodes.
+func AddNextEpochNodes(cID []byte, nodes [][]byte) {
```

Comment:
Can we have just keys there? Everyone knows the network map anyway (can get details based on key). In fact, even a shorter address handle can be used, but this would require a bit more work (so keys are OK).

Reply:
Also, please use more specific types. `hash256` is much better than just `[]byte`.

Reply:
> i think we should clarify what exactly these byte arrays are and verify them 1st

> Also, please use more specific types. hash256 is much better than just []byte.

@roman-khimov, if it is `hash256`, can it be ensured by the VM? or still we need to check it manually?

> Can we have just keys there?

we have discussed it a little and i thought that we make it clear that it is useless for current nodes and IR. if you need a container, then you need some ways to communicate with it, not just know its keys. it can be useful for the future things (https://github.com/nspcc-dev/neofs-contract/issues/413), is this PR for contract-side calculations only?

Reply:
> if it is hash256, can it be ensured by the VM?

No, https://github.com/neo-project/neo/issues/1891.

> if you need a container, then you need some ways to communicate with it

Well, you can get a node by key. And full node info can take quite some space. But let's say it's OK for now.

Reply:
> No, https://github.com/neo-project/neo/issues/1891.

fresh thing. ok, type is type, fixed 

> But let's say it's OK for now.

what exacty OK? this PR or storing keys only? i like storing only keys more. just want to be sure we all understand SN and IR will not be using it in the current code, only for new contract features. they will likely use a single network request, a single calculation (placement) and caching, not this PR

Reply:
If we're targeting meta only then full node info isn't needed. But eventually I think this list can be used by other code since no one cares about placement policy, everyone just needs a node list to work with. Full node info can still be provided if this contract is to store keys only, not immediately, but after `netmap` refactorings. So overall keys only are OK.